homeassistant/components/geniushub/__init__.py

```diff
@@ -13,25 +13,38 @@
 
 DOMAIN = 'geniushub'
 
+_V1_API_SCHEMA = vol.Schema({
+    vol.Required(CONF_TOKEN): cv.string,
```

Comment:
See the temperature set schema in the climate base component. Use that pattern instead.

Reply:
Do you mean: https://github.com/home-assistant/home-assistant/blob/a97fb8fd10af84df15bb952342c1bb050dee6a67/homeassistant/components/climate/__init__.py#L91-L101

I had a look at it, and couldn't see how to get it to work.  I can't see what's wrong with my solution: it's pretty simple, and self-explanatory.

I am happy to make any changes you suggest, but I just can't see what you mean - Could you explain further?

Reply:
```py
SET_TEMPERATURE_SCHEMA = vol.Schema(vol.All( 
	     cv.has_at_least_one_key( 
	         CONF_HOST, CONF_TOKEN), 
	     { 
	         vol.Exclusive(CONF_TOKEN, 'hub_id'): cv.string, 
	         vol.Inclusive(CONF_HOST, 'hub_id'): cv.string, 
	         vol.Inclusive(CONF_USERNAME, 'hub_id'): cv.string, 
	         vol.Inclusive(CONF_PASSWORD, 'hub_id'): cv.string, 
	     } 
	 ))
```

Reply:
This schema is *not* XOR, and will incorrectly accept the following configuration as valid :
```yaml
geniushub:
  token: !secret genius_token
  host: hub.genius.com
  username: !secret genius_username
  password: !secret genius_password
```

My schema will reject the above, and just looks better.